src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -187,10 +188,11 @@ fn generic_extension<'cx>(cx: &'cx mut ExtCtxt,
                     if comma_span == DUMMY_SP {
                         err.note("you might be missing a comma");
                     } else {
-                        err.span_suggestion_short(
-                            comma_span,
-                            "missing comma here",
-                            ", ".to_string(),
+                        err.span_suggestion_short_with_applicability(
+                                                comma_span,
+                                                "missing comma here",
+                                                ", ".to_string(),
+                                                Applicability::MachineApplicable,
```

Comment:
This might be the first syntax error fix we have -- are we sure this is 100% correct in this case, @estebank?

Reply:
I don't think it is the first one, but this suggestion will always be syntactically correct, as in the new code will match one of the macro matches. I can imagine cases where that wouldn't be the expected case, but I'm pretty sure >90% we'll be correct.